src/graph/graphcontext.c

```diff
@@ -161,7 +197,7 @@ int GraphContext_AddIndex(GraphContext *gc, const char *label, const char *attri
   if(idx) return INDEX_FAIL;
 
   // Populate an index for the label-attribute pair using the Graph interfaces.
-  Attribute_ID attr_id = Schema_GetAttributeID(s, attribute);
+  Attribute_ID attr_id = Attribute_GetID(attribute); // TODO silly
```

Comment:
All of this work is already performed once within the above call to `Schema_GetIndex`, the access patterns should be updated so that we only retrieve the ID once.